src/main/java/org/jenkinsci/plugins/workflow/cps/global/CompilationErrorException.java

```diff
@@ -0,0 +1,18 @@
+package org.jenkinsci.plugins.workflow.cps.global;
+
+import org.codehaus.groovy.control.MultipleCompilationErrorsException;
+
+import java.io.NotSerializableException;
+
+/**
+ * An exception that replaces Groovy's {@link MultipleCompilationErrorsException},
+ * because that is not serializable (which would, under certain circumstances,
+ * lead to the user being presented with a {@link NotSerializableException} instead
+ * of a compilation error -- see JENKINS-40109).
+ */
+public class CompilationErrorException extends RuntimeException {
+    public CompilationErrorException(MultipleCompilationErrorsException original) {
+        super(original.toString());
```

Comment:
nit: I'd have to check what the exact output is to know for sure, but my first guess is that it would be better to use `original.getMessage()` instead of `original.toString()`.

Reply:
That is what I meant to call, except I mistyped last second. Nice catch.